cmd/main.go

```diff
@@ -434,7 +436,36 @@ func autoMigrationWithLeader(migrationLeader leader.ElectorInterface, db *gorm.D
 }
 
 func uploadBaseISOWithLeader(uploadLeader leader.ElectorInterface, objectHandler s3wrapper.API, generator generator.ISOInstallConfigGenerator, log logrus.FieldLogger) error {
+
 	ctx := context.Background()
+
+	// First, we must determine our base object
+	if Options.RHCOSBaseImage == "" {
+		s3wrapper.BaseObjectName = s3wrapper.RHCOSBaseObjectName
+	} else {
+		// Need to get the volume id from the ISO provided
+		iso, err := os.Open(Options.RHCOSBaseImage)
+		if err != nil {
+			log.WithError(err).Fatal("Failed to open provided base image for inspection")
+			return err
+		}
+		defer iso.Close()
+
+		// Need a method to identify the ISO provided
+		// The first 32768 bytes are unused by the ISO 9660 standard, typically for bootable media
+		// This is where the data area begins and the 32 byte string representing the volume identifier
+		// is offset 40 bytes into the primary volume descriptor
+		volumeId := make([]byte, 32)
+		_, err = iso.ReadAt(volumeId, 32808)
+		if err != nil {
+			log.WithError(err).Fatal("Failed to read volume id from provided base image")
+			return err
+		}
+		// TODO: Should we verify that the volume id is of the form `rhcos-<version>`?
+		s3wrapper.BaseObjectName = strings.TrimSpace(string(volumeId)) + ".iso"
+	}
+
+	// Second, we check if it is already in s3
 	exists, err := objectHandler.DoesObjectExist(ctx, s3wrapper.BaseObjectName)
```

Comment:
Should we move this check into the `RunWithLeader` block?
Otherwise each replica could see this as `false` and then sequentially upload the file multiple times as each acquires the leader lock.